### ISSUE TITLE
`Popover` is missing `dialog` role

### ISSUE THREAD

**Issue by jasonjwichert** [iTwin/stratakit#1421]
The `Popover` component should have a `dialog` role by default. The [example code](https://github.com/iTwin/stratakit/blob/37034c7ca2a4361619abfeb136d8666928a01f1b/examples/mui/Popover.default.tsx#L33) adds it currently, but we should automatically add it. The example adds it to the main `<Popover>` component, however I would argue that it should be in `slotProps` on the `paper` component, as the main component includes the invisible backdrop.

**Comment by mayank99** [comment]
> The example adds it to the main `<Popover>` component, however I would argue that it should be in `slotProps` on the `paper` component, as the main component includes the invisible backdrop.

While the `role` can be set on any of the elements from the outermost to the innermost wrapper, what's important is the element with the `role="dialog"` is the one that is being labelled (using `aria-labelledby` in the example).

There is also `aria-controls` pointing to the `role="dialog"` element in the example code, though I think we can  remove that due to lack of support.

**Comment by jasonjwichert** [comment]
Updated the example code in #1420 to move `aria-labelledby` to the `paper`'s `slotProps`, IMO we should definitely have the `dialog` role on the paper as this is the pattern that `Dialog` follows

**Comment by jasonjwichert** [comment]
Open to discussion though as that makes the manual props that the users have to add for the trigger/popover association more complicated

**Comment by mayank99** [comment]
> that makes the manual props that the users have to add for the trigger/popover association more complicated

This was my concern as well. Ideally the consumer doesn't need to understand the internal implementation of the component. But we do already add a class to the `paper` slot for padding, so maybe this argument is moot.

**Comment by jasonjwichert** [comment]
@GerardasB @FlyersPh9 any thoughts on this? 

TLDR: to follow the conventions that MUI's `Dialog` already uses, my PR put `role="dialog"` on `Popover`'s `paper` slot instead of the root component. This makes it more complicated for SK consumers to add `aria-labelledby` as they have to add it to the `paper` slot prop instead of just on the `Popover`

I guess it depends if we want to follow MUI's `Dialog` conventions where the root has `role="presentation"` and the `paper` slot has `role="dialog"`, or if we just want to put `role="dialog"` on the `Popover` root for ease of use.

It did seem odd to me that `Popover` even has a `backdrop` since it should always be transparent anyway, so maybe we don't need to preserve the `presentation` → `dialog` role hierarchy?

**Comment by GerardasB** [comment]
@jasonjwichert I agree, that we should keep consistent with the `Dialog` and put the `role="dialog"` on the `paper` slot.

However, in terms of API usage as proposed by #1420, there is a difference/discrepancy:
`Dialog` component passes down certain props to the "paper", rather than the "root" element, see:
https://github.com/mui/material-ui/blob/631d685ec83a369c012ccc889787ee380d74b6c3/packages/mui-material/src/Dialog/Dialog.js#L360-L364
This is visible in the examples: https://mui.com/material-ui/react-dialog/#alerts

Maybe we should (if possible), pass-down `aria-label`/`aria-labelledby` props set on `Popover` to the `paper` slot automagically?

**Comment by mayank99** [comment]
> pass-down `aria-label`/`aria-labelledby` props set on `Popover` to the `paper` slot automagically?

I don't think we should do this. It adds complexity while making things more difficult to understand (e.g. `aria-labelledby` delegated to `paper`, but other `aria-*` would remain on root?)

If we want to improve the DX by accepting top-level `aria-labelledby`, then we should apply `role="dialog"` to the root slot. It's simpler and more predictable. Or we could keep the current approach of the PR, where `aria-labelledby` is passed into `slotProps.paper` (and documented).

**Comment by jasonjwichert** [comment]
Will let you guys make the final call, but IMO it makes more sense to have `dialog` on the paper itself and the root that includes the invisible backdrop, and if our example and docs are clear DX shouldn't be an issue even though it's a little less clean

**Comment by GerardasB** [comment]
> Will let you guys make the final call, but IMO it makes more sense to have `dialog` on the paper itself and the root that includes the invisible backdrop, and if our example and docs are clear DX shouldn't be an issue even though it's a little less clean

Confirming, that we have agreed to keep `role="dialog"` on the `paper` slot and "require" the label via the `paper` slot.

**Comment by mayank99** [comment]
This is now released in [0.4.1](https://github.com/iTwin/stratakit/releases/tag/%40stratakit%2Fmui%400.4.1). Thanks again for your contribution!
